include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
+        std::string str = std::to_string(t);
+        str.erase(str.find_last_not_of('0') + 1, std::string::npos);
+        if (str.back() == '.') {
```

Comment:
One data point: iOS system region currently has no effect on what decimal separator is used here. The separator is still a decimal point with the region set to German, which is “4.567,89” for 4,567.89 in the US.